File: Juliandra_flutter/lib/models/agenda.dart
class Agenda {
  final int id;
  final String judul;
  final String? keterangan;
  final bool isDone;
  final DateTime createdAt;
  final DateTime updatedAt;

  Agenda({
    required this.id,
    required this.judul,
    this.keterangan,
    required this.isDone,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Agenda.fromJson(Map<String, dynamic> json) {
    return Agenda(
      id: json['id'],
      judul: json['judul'],
      keterangan: json['keterangan'],
      isDone: json['is_done'] == 1 || json['is_done'] == true,
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'judul': judul,
      'keterangan': keterangan,
      'is_done': isDone ? 1 : 0,
      'created_at': createdAt.toIso8601String(),
      'updated_at': updatedAt.toIso8601String(),
    };
  }

  Map<String, dynamic> toCreateJson() {
    return {
      'judul': judul,
      'keterangan': keterangan,
      'is_done': isDone,
    };
  }

  Map<String, dynamic> toUpdateJson() {
    return {
      'judul': judul,
      'keterangan': keterangan,
      'is_done': isDone,
    };
  }
}

File: Juliandra_api/app/Models/Agenda.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Agenda extends Model
{
    /** @use HasFactory<\Database\Factories\AgendaFactory> */
    use HasFactory;

    /**
     * The table associated with the model.
     *
     * @var string
     */
    protected $table = 'agenda';

    /**
     * The attributes that are mass assignable.
     *
     * @var list<string>
     */
    protected $fillable = [
        'judul',
        'keterangan',
        'is_done',
    ];

    /**
     * Get the attributes that should be cast.
     *
     * @return array<string, string>
     */
    protected function casts(): array
    {
        return [
            'is_done' => 'boolean',
            'created_at' => 'datetime',
            'updated_at' => 'datetime',
        ];
    }
}

